packages/camera/camera/example/integration_test/camera_test.dart

```diff
@@ -240,4 +240,34 @@ void main() {
     },
     skip: !Platform.isAndroid,
   );
+
+  testWidgets(
+    'Image streaming persistence test on iOS',
```

Comment:
Would this test fail if the rest of the PR were reverted? It doesn't seem that way, unless I'm missing something about the test.

Reply:
It is a device dependent test, so if you test it on an old device, it will crash.

Reply:
This means that it wouldn't actually fail in our CI, which means there's nothing preventing this from breaking again.

A test for something like this should be creating the conditions to allow for testing that the behavior is what we want regardless of the device, so that it has predictable output. If the goal is to prevent more than N frames from being in flight without response, for instance, then a test should deliberately respond very slowly and assert that N isn't exceeded.